pkg/admission/namespacelifecycle/admission_test.go

```diff
@@ -18,136 +18,66 @@ package namespacelifecycle
 
 import (
 	"context"
-	"fmt"
 	"testing"
-	"time"
 
 	"github.com/kcp-dev/logicalcluster/v2"
 
 	corev1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apiserver/pkg/admission"
-	kubeadmission "k8s.io/apiserver/pkg/admission/initializer"
 	"k8s.io/apiserver/pkg/endpoints/request"
-	informers "k8s.io/client-go/informers"
-	clientset "k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/kubernetes/fake"
-	core "k8s.io/client-go/testing"
 
 	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
 )
 
 // newHandlerForTestWithClock returns a configured handler for testing.
-func newHandlerForNamespaceLifeCycle(kubeClient clientset.Interface, lister fakeClusterWorkspaceLister) (*workspaceNamespaceLifecycle, error) {
-	f := informers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
+func newHandlerForNamespaceLifeCycle(workspace *tenancyv1alpha1.ClusterWorkspace) (*workspaceNamespaceLifecycle, error) {
```

Comment:
Could we just inline this to the test? The godoc already drifted and it's not clear this is reusable in any meaningful way

Reply:
Done